api/v1alpha2/source_types.go

```diff
@@ -0,0 +1,104 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// BundleDeploymentSource configures the sources
+// from where the bundle content needs to be
+// downloaded, and the location to unpack it.
+type BundleDeplopymentSource struct {
+	// Image is the bundle image that backs the content of this bundle.
+	Image *ImageSource `json:"image,omitempty"`
+
+	// Git is the git repository that backs the content of this Bundle.
+	Git *GitSource `json:"git,omitempty"`
+
+	//  HTTP is the remote location that backs the content of this Bundle.
+	HTTP *HTTPSource `json:"http,omitempty"`
+
+	// Relative location to place the fetched artifacts
+	// +optional
+	Destination string `json:"destination,omitempty"`
+}
+
+type ImageSource struct {
+	// Ref contains the reference to a container image containing Bundle contents.
+	Ref string `json:"ref"`
+
+	// Auth configures the authentication method if necessary.
+	Auth *Authentication `json:"auth,omitempty"`
+}
+
+type GitSource struct {
+	// Repository is a URL link to the git repository containing the bundle.
+	// Repository is required and the URL should be parsable by a standard git tool.
+	Repository string `json:"repository"`
+
+	// Directory refers to the location of the bundle within the git repository.
+	// Directory is optional and if not set defaults to `manifests`.
+	//
+	// +kubebuilder:default:=manifests
+	Directory string `json:"directory,omitempty"`
+
+	// Ref configures the git source to clone a specific branch, tag, or commit
+	// from the specified repo. Ref is required, and exactly one field within Ref
+	// is required. Setting more than one field or zero fields will result in an
+	// error.
+	Ref GitRef `json:"ref"`
+
+	// Auth configures the authentication method if necessary.
+	Auth *Authentication `json:"auth,omitempty"`
+}
+
+type GitRef struct {
+	// Branch refers to the branch to checkout from the repository.
+	// The Branch should contain the bundle manifests in the specified directory.
+	Branch string `json:"branch,omitempty"`
+
+	// Tag refers to the tag to checkout from the repository.
+	// The Tag should contain the bundle manifests in the specified directory.
+	Tag string `json:"tag,omitempty"`
+
+	// Commit refers to the commit to checkout from the repository.
+	// The Commit should contain the bundle manifests in the specified directory.
+	Commit string `json:"commit,omitempty"`
+}
+
+type Authentication struct {
+	// Secret contains reference to the secret that has authorization information and is in the namespace that the provisioner is deployed.
+	// The secret is expected to contain `data.username` and `data.password` for the username and password, respectively for http(s) scheme.
```

Comment:
Agree, but the alpha1 API currently supports authentication only using secrets that contain username and password for image and http based source. I can create an issue to track this, so that we can handle it in follow ups.